lib/bundler/definition.rb

```diff
@@ -184,11 +184,10 @@ def specs_for(groups)
     # @return [SpecSet] resolved dependencies
     def resolve
       @resolve ||= begin
+        last_resolve = converge_locked_specs
```

Comment:
Moving this out is a bigger behavioural change than I would have liked, but it was the smallest code change I could work out. All of the tests pass (locally)... is there anything else this could break?